src/db/emoji_logging.rs

```diff
@@ -97,22 +95,28 @@ impl Db {
     pub async fn get_top_emoji_stats(
         &self,
         count: u16,
-    ) -> Result<impl Iterator<Item = EmojiStats>> {
+        ordering: Ordering,
+    ) -> Result<Box<dyn Iterator<Item = EmojiStats> + Send + Sync>> {
         let mut conn = self.pool.acquire().await?;
-        let records = sqlx::query!(
+        let records = sqlx::query_unchecked!(
             r#"select *, in_text_usage + reaction_usage as "usage!: i32" FROM emoji_stats order by "usage!: i32" DESC limit ?"#,
             count
         )
         .fetch_all(&mut conn)
         .await?;
-        Ok(records.into_iter().map(|x| EmojiStats {
+
+        let final_results = records.into_iter().map(|x| EmojiStats {
             emoji: EmojiIdentifier {
                 id: EmojiId(x.emoji_id as u64),
                 animated: x.animated != 0,
                 name: x.emoji_name.unwrap(),
             },
             in_text: x.in_text_usage as u64,
             reactions: x.reaction_usage as u64,
-        }))
+        });
+        match ordering {
+            Ordering::Descending => Ok(Box::new(final_results)),
+            Ordering::Ascending => Ok(Box::new(final_results.rev())),
```

Comment:
ordering should be done in SQL, by changing DESC to ASC, not in rust

Reply:
impossible. otherwise i have to match the ordering and reconstruct the struct because the queries are two completely separate typed and you cant pass them as paremeters as far as im aware

Reply:
and reverse is the least of our worries as the sorting is already done on the sql side

Reply:
all this does is show you the top 10 emojis in reverse order. I don't think that's what @6gk wants. It's not impossible, just do something like 
```rust
let records = match ordering {
  Descending => sqlx::query!(
              r#"select *, in_text_usage + reaction_usage as "usage!: i32" FROM emoji_stats order by "usage!: i32" DESC limit ?"#,
              count
          )
          .fetch_all(&mut conn)
          .await?
  }
  Ascending => sqlx::query!(
              r#"select *, in_text_usage + reaction_usage as "usage!: i32" FROM emoji_stats order by "usage!: i32" ASC limit ?"#,
              count
          )
          .fetch_all(&mut conn)
          .await?
};
```

Reply:
or if the problem is that the type will different in both of these branches then convert them to EmojiStats in each branch

Reply:
maybe make a macro that takes in a query and does .fetch_all, .await and conversion to EmojiStats